homeassistant/components/vulcan/calendar.py

```diff
@@ -145,53 +111,44 @@ async def async_get_events(self, hass, start_date, end_date):
             _LOGGER.error("An API error has occurred: %s", err)
             events = []
         except ClientConnectorError as err:
-            if self._available:
+            if self.service_available:
                 _LOGGER.warning(
                     "Connection error - please check your internet connection: %s", err
                 )
             events = []
 
         event_list = []
         for item in events:
-            event = {
-                "uid": item["id"],
-                "start": {
-                    "dateTime": datetime.combine(
-                        item["date"], item["time"].from_
-                    ).strftime(DATE_STR_FORMAT)
-                },
-                "end": {
-                    "dateTime": datetime.combine(
-                        item["date"], item["time"].to
-                    ).strftime(DATE_STR_FORMAT)
-                },
-                "summary": item["lesson"],
-                "location": item["room"],
-                "description": item["teacher"],
-            }
+            event = CalendarEvent(
+                start=datetime.combine(item["date"], item["time"].from_),
+                end=datetime.combine(item["date"], item["time"].to),
+                summary=item["lesson"],
+                location=item["room"],
+                description=item["teacher"],
+            )
 
             event_list.append(event)
 
         return event_list
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
```

Comment:
Side note: Please describe the purpose of the throttle. Now each entity has its own throttle and all entities share the same client, so the throttle will not throttle requests per client.